File: backEnd/family-care-clinic/src/main/java/com/ninos/patient/service/PatientService.java
package com.ninos.patient.service;

import com.ninos.enums.BloodGroup;
import com.ninos.enums.Genotype;
import com.ninos.patient.dto.PatientDTO;
import com.ninos.res.Response;

import java.util.List;

public interface PatientService {

    Response<PatientDTO> getPatientProfile();
    Response<?> updatePatientProfile(PatientDTO patientDTO);
    Response<PatientDTO> getPatientById(Long patientId);
    Response<List<BloodGroup>> getAllBloodGroupEnums();
    Response<List<Genotype>> getAllGenotypeEnums();


}

File: frontEnd/family-care-clinic-ui/src/app/services/patient-service.ts
import { Injectable } from '@angular/core';
import { AuthService } from './auth-service';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class PatientService {

  private BASE_URL = environment.BASE_URL;

  constructor(private http: HttpClient, private authService: AuthService) { }

  getPatientProfile(): Observable<any> {
    return this.http.get(`${this.BASE_URL}/patients/me`,
      { headers: this.authService.getHeader() });
  }

  updatePatientProfile(body: any): Observable<any> {
    return this.http.put(`${this.BASE_URL}/patients/me`, body,
      { headers: this.authService.getHeader() });
  }

  getPatientById(patientId: number | string): Observable<any> {
    return this.http.get(`${this.BASE_URL}/patients/${patientId}`,
      { headers: this.authService.getHeader() });
  }

  getAllGenotypeEnums(): Observable<any> {
    return this.http.get(`${this.BASE_URL}/patients/genotype`,
      { headers: this.authService.getHeader() });
  }

  getAllBloodGroupEnums(): Observable<any> {
    return this.http.get(`${this.BASE_URL}/patients/blood-group`,
      { headers: this.authService.getHeader() });
  }








}
